src/umap.cpp

```diff
@@ -153,7 +128,8 @@ void* umap_fits(void* addr, size_t length, int prot, int flags, int fd_num,int*
 
     flags |= (MAP_ANONYMOUS | MAP_NORESERVE);
 
-    void* region = mmap(addr, length, prot, flags, -1, 0);
+    void* region = (fd_list == NULL)? mmap(addr, length, prot, flags, -1, offset):mmap(addr, length, prot, flags, -1, 0);
```

Comment:
The umap implementation works as follows:
- It makes an MAP_ANONYMOUS mmap() call (no backing file, no offset into the file).  By definition, the file descriptor is specified to -1 and the offset 'is ignored'.  Because as far as mmap is concerned, there is no backing file.
- It then manages a buffer of pages and implements its own backing file mechanism using the passed in file descriptor(s) and offset(s)